src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplamListServiceImpl.java

```diff
@@ -123,6 +158,157 @@ public GermplasmList getGermplasmList(final Integer germplasmListId) {
+		this.validateListName(currentProgram, name);
+
+		final String parentFolderId = request.getParentFolderId();
+		checkNotNull(parentFolderId, "param.null", new String[] {"parentFolderId"});
+		this.validateParentId(parentFolderId, currentProgram);
+
+		// process and assign defaults + more validations
+		this.processEntries(request.getEntries());
+
+		// properties that depend on CROP/PROGRAM folder
+		int status = GermplasmList.Status.LIST.getCode();
+		String programUUID = null;
+		// If the germplasm list is saved in 'Crop lists' folder, the programUUID should be null
+		// so that the germplasm list will be accessible to all programs of the same crop.
+		if (CROP_LISTS.equals(parentFolderId)) {
+			// list should be locked by default if it is saved in 'Crop lists' folder.
+			status = GermplasmList.Status.LOCKED_LIST.getCode();
+		} else {
+			programUUID = currentProgram;
+		}
+
+		if (CROP_LISTS.equals(parentFolderId) || PROGRAM_LISTS.equals(parentFolderId)) {
+			request.setParentFolderId(null);
+		}
+
+		final WorkbenchUser loggedInUser = this.securityService.getCurrentlyLoggedInUser();
+
+		// finally save
+		return this.germplasmListService.create(request, status, programUUID, loggedInUser);
+	}
+
+	private void processEntries(final List<GermplasmListGeneratorDTO.GermplasmEntryDTO> entries) {
+
+		final List<Integer> gids = entries.stream().map(GermplasmListGeneratorDTO.GermplasmEntryDTO::getGid).collect(Collectors.toList());
+		final Map<Integer, String> preferrredNamesMap = this.germplasmDataManager.getPreferredNamesByGids(gids);
+		final Map<Integer, Germplasm> germplasmMap = this.germplasmDataManager.getGermplasms(gids)
+			.stream().collect(Collectors.toMap(Germplasm::getGid, Function.identity()));
+
+		int entryNo = 1;
+		boolean hasEntryNo = false;
+		boolean hasEntryCode = false;
+		boolean hasEntryCodeEmpty = false;
+		boolean hasSeedSource = false;
+		boolean hasSeedSourceEmpty = false;
+		boolean hasDesignation = false;
+		boolean hasDesignationEmpty = false;
+		boolean hasGroupName = false;
+		boolean hasGroupNameEmpty = false;
+
+		for (final GermplasmListGeneratorDTO.GermplasmEntryDTO entry : entries) {
+			final Integer gid = entry.getGid();
+			if (gid == null) {
+				throw new ApiValidationException("", "error.germplasmlist.save.gid");
+			} else if (germplasmMap.get(gid) == null) {
+				throw new ApiValidationException("", "error.germplasmlist.save.gid.not.exists", gid);
+			}
+
+			if (entry.getEntryNo() == null) {
+				entry.setEntryNo(entryNo++);
+			} else {
+				hasEntryNo = true;
+			}
+
+			if (isBlank(entry.getEntryCode())) {
+				entry.setEntryCode(String.valueOf(entry.getEntryNo()));
+				hasEntryCodeEmpty = true;
+			} else {
+				hasEntryCode = true;
+			}
+
+			if (isBlank(entry.getSeedSource())) {
+				entry.setSeedSource(GermplasmListDataDAO.SOURCE_UNKNOWN);
+				hasSeedSourceEmpty = true;
+			} else {
+				hasSeedSource = true;
+			}
+
+			if (isBlank(entry.getDesignation())) {
```

Comment:
Same question here, I think we should not even consider the Designation to be set by the users if the service is used by an external client

Reply:
I'll remove it